yarn-project/acir-simulator/src/avm/opcodes/comparators.ts

```diff
@@ -1,22 +1,25 @@
 import { AvmMachineState } from '../avm_machine_state.js';
-import { Field } from '../avm_memory_types.js';
+import { TypeTag } from '../avm_memory_types.js';
 import { AvmJournal } from '../journal/index.js';
 import { Instruction } from './instruction.js';
 
 export class Eq extends Instruction {
   static type: string = 'EQ';
   static numberOfOperands = 3;
 
-  constructor(private aOffset: number, private bOffset: number, private destOffset: number) {
+  constructor(private aOffset: number, private bOffset: number, private dstOffset: number, private inTag: TypeTag) {
```

Comment:
I think it makes sense for flags/tags to come before the operands in the constructor. Thoughts? They will come directly after the opcode in the bytecode.

Reply:
Ah, yes, this is a thing. My rationale was that actually the most important parts are the operands. Like when you say "ADD<t> A B C" a, b, c are the most important for readability. The flags are kind of secondary.

Having said that (1) I have no strong feelings but (2) it's already like this in all other opcodes that support inTag, so I'd rather keep it consistent :) we can change them all together.